test/test_dos.py

```diff
@@ -4485,6 +4486,74 @@ def test_lfn_file_info_mfs_6GiB(self):
         """LFN file info on MFS (6 GiB)"""
         self._test_lfn_file_info_mfs(1024 * 1024 * 1024 * 6)
 
+    def _test_cpu(self, dpmitype):
+        mkfile("testit.bat", """\
+test-i386 > test.log\r
+rem end\r
+""")
+
+        symlink("../../../src/tests/test-i386.exe",
+                "test-imagedir/dXXXXs/c/test-i386.exe")
+
+        parameters = {
+            "vm86native": ("vm86", "native", "off"),
+            "kvmnative": ("kvm", "native", "off"),
+            "jitnative": ("emulated", "native", "vm86"),
+            "simnative": ("emulated", "native", "vm86sim"),
+            "kvm": ("kvm", "kvm", "off"),
+            "jit": ("emulated", "emulated", "full"),
+            "sim": ("emulated", "emulated", "fullsim")}[dpmitype]
+        results = self.runDosemu("testit.bat", config="""\
+$_hdimage = "dXXXXs/c:hdtype1 +1"
+$_floppy_a = ""
+$_cpu_vm = "%s"
+$_cpu_vm_dpmi = "%s"
+$_cpu_emu = "%s"
+$_ignore_djgpp_null_derefs = (off)
+"""%parameters)
+
+        with open("test-imagedir/dXXXXs/c/test.log") as f:
+            lines = list(f)
+            self.assertEqual(len(lines), 5056)
+            # compare or copy to reference file
+            try:
+                with open("test-i386.log") as g:
+                    self.maxDiff = None
+                    self.assertEqual(list(g), lines)
+            except IOError:
+                # copy as reference file
+                with open("test-i386.log", "w") as g:
+                    g.write("".join(lines))
+
+    def test_cpu_1_vm86native(self):
+        """CPU test: native vm86 + native DPMI (i386 only)"""
+        if uname()[4] == 'i686':
+            self._test_cpu("vm86native")
```

Comment:
Perhaps you could revise this to 
```python
if uname()[4] == 'x86_64':
   self.skipTest("x86_64 doesn't support native vm86()")
self._test_cpu("vm86native")
```

Other than that, it looks great, I'm so glad you done this!